extensions/sockets/src/main/java/org/mule/module/socket/internal/DelegatingInputStream.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class DelegatingInputStream extends InputStream
```

Comment:
apache commons already has classes like this. Actually, you might one to take a look at AutoCloseInputStream